### PR TITLE
Get NUnit test running/debugging working

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/1434

This change updates the test manager to pass dummy `RunSettings`, which allows the latest NUnit alpha to work (the NUnit test adapter does not handle null RunSettings). In addition, this updates our NUnit test project to use the latest NUnitTestAdapter CI build and uncomments the NUnit tests. When there's an official NUnitTestAdapter release, we'll want to update the test project to reference that and get rid of its NuGet.config.

cc @bernardbr and @rprouse

### CODE REVIEW

File: test-assets/test-projects/NUnitTestProject/NuGet.config [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="AppVeyor NUnit CI Feed" value="https://ci.appveyor.com/nuget/nunit" />
```
Comment (rprouse) [inline test-assets/test-projects/NUnitTestProject/NuGet.config]: Minor, but you don't need this line/feed unless you want to pull in pre-release versions of the NUnit Framework. The adapter PR only uses the Engine and Adapter feeds.

File: tests/OmniSharp.DotNetTest.Tests/RunTestFacts.cs [hunk 2]
```diff
@@ -65,10 +65,7 @@ public RunTestFacts(ITestOutputHelper testOutput)
             Assert.Equal(1, response.Results.Length);
         }
 
-        // NUnit does not work with .NET CLI RTM yet. https://github.com/nunit/dotnet-test-nunit/issues/108
-        // When it does, the NUnitTestProject should be updated and the tests below re-enabled.
-
-        //[Fact]
+        [Fact]
```
Comment (rprouse) [inline tests/OmniSharp.DotNetTest.Tests/RunTestFacts.cs]: Testing NUnit with xUnit, ironic 😄

**Comment by rprouse** [comment]
Looks good. At my end, I will push an update to the adapter to handle null `RunSettings`.

File: tests/OmniSharp.DotNetTest.Tests/RunTestFacts.cs (hunk 2 above)
Comment (DustinCampbell) [inline tests/OmniSharp.DotNetTest.Tests/RunTestFacts.cs]: well, NUnit doesn't work yet, so...

File: test-assets/test-projects/NUnitTestProject/NuGet.config (hunk 1 above)
Comment (DustinCampbell) [inline test-assets/test-projects/NUnitTestProject/NuGet.config]: As soon as there's an NUnitTestAdapter on nuget.org that we can use, I'll just delete this file.

**Comment by bernardbr** [comment]
Thank you very much @DustinCampbell and @rprouse !

Noob's Question: How can I use this version of omni-sharp? :smile:

**Comment by DustinCampbell** [comment]
@bernardbr: I'll be producing a new omnisharp-vscode beta later today with this fix.

**Comment by bernardbr** [comment]
Ok @DustinCampbell 
Thanks again!

**Comment by DustinCampbell** [comment]
Alright. CI passed again. I'm going to go ahead and merge.

**Comment by rprouse** [comment]
@bernardbr I fixed this in the NUnit adapter also with nunit/nunit3-vs-adapter@d345c72 and the NuGet package 4.0.0-ci-00455-pr-313 and newer should work without these fixes.

**Comment by bernardbr** [comment]
Hello @rprouse!
Can you help me to see what I did wrong?
I do the update of Test Adapter for the version you've mentioned, but the "null" error still occurring!
[UnitTest.zip](https://github.com/OmniSharp/omnisharp-roslyn/files/964326/UnitTest.zip)

Thanks!

**Comment by rprouse** [comment]
@bernardbr It isn't working for me either. I am attempting to debug and will report back.

**Comment by DustinCampbell** [comment]
Try the 1.10.0-beta1 release of C# for VS Code I created last night. Follow these steps [Installing Beta Releases](https://github.com/OmniSharp/omnisharp-vscode/wiki/Installing-Beta-Releases).

**Comment by rprouse** [comment]
@DustinCampbell your fix is working. I just can't figure out why my fix isn't working and I am having trouble debugging with all the layers. What is troubling me is that I believe your fix confirmed my diagnosis that we weren't handling a null `RunSettings` and I switched our code to use defaults if it is null or empty along with unit tests to prove that it works. I've spent the last hour digging through our code, but there are no other usages. My test is effectively the same as yours, just at our end.

Is there anyway to get the callstack like you displayed in the issue without a debug compile?

image

**Comment by DustinCampbell** [comment]
Not without a compile of OmniSharp without my change. That's not actually too big of a deal. I usually do it like so on Windows:

1. Get latest from omnisharp-roslyn dev branch and run `build.cmd`
2. Open OmniSharp.sln in VS 2017 and build
3. In VS Code, select Preferences->User Settings from the menu and add the following setting to the settings.json:

```JSON
"omnisharp.path": "<path to your omnisharp folder>/OmniSharp/bin/Debug/net46/OmniSharp.exe"
```

4. Open your test project in VS Code.

**Pull request merged**